src/hooks/useAppBootstrap.js

```diff
@@ -1,72 +1,24 @@
-import Minilog from '@cozy/minilog'
 import { Linking, LogBox } from 'react-native'
 import { useEffect, useState } from 'react'
+import { deconstructCozyWebLinkWithSlug } from 'cozy-client'
 
 import { SentryCustomTags, setSentryTag } from '/libs/monitoring/Sentry'
 import { devConfig } from '/constants/dev-config'
 import { manageIconCache } from '/libs/functions/iconTable'
+import { getDefaultIconParams } from '/libs/functions/openApp'
 import { navigate } from '/libs/RootNavigation'
 import { routes } from '/constants/routes'
 import { useSplashScreen } from '/hooks/useSplashScreen'
 
-if (devConfig.ignoreLogBox) LogBox.ignoreAllLogs()
-
-const log = Minilog('useAppBootstrap')
-
-const parseOnboardingURL = url => {
-  try {
-    if (!url || !url.includes('onboarding')) {
-      return undefined
-    }
-
-    const onboardingUrl = new URL(url)
-    const onboardUrl = onboardingUrl.searchParams.get('onboard_url')
-    const fqdn = onboardingUrl.searchParams.get('fqdn')
-
-    if (!onboardUrl && !fqdn) {
-      return undefined
-    }
-
-    return {
-      onboardUrl,
-      fqdn
-    }
-  } catch (error) {
-    log.error(
-      `Something went wrong while trying to parse onboarding URL data: ${error.message}`
-    )
-    return undefined
-  }
-}
-
-const parseFallbackURL = url => {
-  const defaultParse = { fallback: undefined, root: routes.stack }
-
-  if (url === null) {
-    return defaultParse
-  }
-
-  try {
-    const makeURL = new URL(url)
-    const fallback = makeURL.searchParams.get('fallback')
-    const isHome = makeURL.pathname.split('/')[1] === 'home'
+import {
+  parseFallbackURL,
+  parseOnboardingURL
+} from './useAppBootstrap.functions'
 
-    return {
-      fallback: fallback ? fallback : undefined,
-      root: isHome || !fallback ? routes.stack : routes.cozyapp,
-      isHome
-    }
-  } catch (error) {
-    log.error(
-      `Something went wrong while trying to parse fallback URL data: ${error.message}`
-    )
-    return defaultParse
-  }
-}
+if (devConfig.ignoreLogBox) LogBox.ignoreAllLogs()
 
 export const useAppBootstrap = client => {
```

Comment:
just thinking: at some point we should get rid of `useAppBootstrap` and separate the logic like you did in this PR into multiple modules that we can compose in a lifecycle app hook or even during runtime after initial boot if needed

Reply:
Yes, now that I extracted the URL parse logic it would be easy to split this into 2 hooks, one for the bootstrap on cold start, one for the URL interception on runtime.

However I'm not sure if the hideSplashScreen part can be easily removed from the bootstrap hook.

Reply:
I propose to do this in another PR if you are ok

Reply:
Yes it was definitely for another PR